src/openqaoa-core/backends/qaoa_backend.py

```diff
@@ -1,42 +1,44 @@
 from typing import Union, Optional, List
 import numpy as np
 
+from .plugin_finder import backend_boolean_list
 from .qaoa_vectorized import QAOAvectorizedBackendSimulator
 from .qaoa_analytical_sim import QAOABackendAnalyticalSimulator
 from .devices_core import DeviceBase, DeviceLocal
 from .basebackend import QuantumCircuitBase, QAOABaseBackend
 from ..qaoa_components import QAOADescriptor
-from openqaoa_braket.backends import DeviceAWS, QAOAAWSQPUBackend
-from openqaoa_qiskit.backends import (
-    DeviceQiskit,
-    QAOAQiskitQPUBackend,
-    QAOAQiskitBackendShotBasedSimulator,
-    QAOAQiskitBackendStatevecSimulator,
-)
-from openqaoa_azure.backends import DeviceAzure
-from openqaoa_pyquil.backends import (
-    DevicePyquil,
-    QAOAPyQuilQPUBackend,
-    QAOAPyQuilWavefunctionSimulatorBackend,
-)
-
-
-DEVICE_NAME_TO_OBJECT_MAPPER = {
-    "qiskit.qasm_simulator": QAOAQiskitBackendShotBasedSimulator,
-    "qiskit.shot_simulator": QAOAQiskitBackendShotBasedSimulator,
-    "qiskit.statevector_simulator": QAOAQiskitBackendStatevecSimulator,
-    "vectorized": QAOAvectorizedBackendSimulator,
-    "pyquil.statevector_simulator": QAOAPyQuilWavefunctionSimulatorBackend,
-    "analytical_simulator": QAOABackendAnalyticalSimulator,
-}
-
-DEVICE_ACCESS_OBJECT_MAPPER = {
-    DeviceQiskit: QAOAQiskitQPUBackend,
-    DevicePyquil: QAOAPyQuilQPUBackend,
-    DeviceAWS: QAOAAWSQPUBackend,
-    DeviceAzure: QAOAQiskitQPUBackend,
-}
 
+BACKEND_BOOLEAN_LIST = backend_boolean_list()
+
+def _create_mappers(input_boolean_list: list) -> dict:
+    
+    DEVICE_NAME_TO_OBJECT_MAPPER = dict()
+    DEVICE_ACCESS_OBJECT_MAPPER = dict()
+    
+    DEVICE_NAME_TO_OBJECT_MAPPER["vectorized"] = QAOAvectorizedBackendSimulator
+    DEVICE_NAME_TO_OBJECT_MAPPER["analytical_simulator"] = QAOABackendAnalyticalSimulator
+    
+    if BACKEND_BOOLEAN_LIST[0]:
```

Comment:
I think from here to line ~39 it should all be done programmatically. Using things like `BACKEND_BOOLEAN_LIST[2]` and associating it strictly yo, say `qiskit`, may cause confusion if the order of plugins is changed

Reply:
Agreed. I can change this to a more verbose/strict data structure like a dictionary.